core-java-modules/core-java-streams-4/src/test/java/splitting/StreamSplittingUnitTest.java

```diff
@@ -0,0 +1,96 @@
+import static org.junit.Assert.assertTrue;
+
+public class StreamSplittingUnitTest {
+    private final static List<Article> articles = Lists.newArrayList(
+            new Article("Baeldung", true),
+            new Article("Baeldung", false),
+            new Article("Programming Daily", false),
+            new Article("The Code", false)
+    );
+
+    @Test
+    public void givenListOfArticles_shouldSplitListInTwoCategories_usingPartitioningBy() {
+        Map<Boolean, List<Article>> groupedArticles = articles.stream().collect(Collectors.partitioningBy(a -> a.getTarget().equals("Baeldung")));
+
+        assertTrue(groupedArticles.get(true).containsAll(Lists.newArrayList(
+                new Article("Baeldung", true),
+                new Article("Baeldung", false))));
+        assertTrue(groupedArticles.get(false).containsAll(Lists.newArrayList(
+                new Article("Programming Daily", false),
+                new Article("The Code", false))));
+    }
+
+    @Test
+    public void givenListOfArticles_shouldSplitListInMultipleCategories_usingGroupingBy() {
+        Map<String, List<Article>> groupedArticles = articles.stream().collect(Collectors.groupingBy(Article::getTarget));
+
+        assertEquals(2, groupedArticles.get("Baeldung").size());
+        assertEquals(1, groupedArticles.get("Programming Daily").size());
+        assertEquals(1, groupedArticles.get("The Code").size());
+
+        assertTrue(groupedArticles.get("Baeldung").containsAll(Lists.newArrayList(
+                new Article("Baeldung", true),
+                new Article("Baeldung", false))));
+        assertTrue(groupedArticles.get("Programming Daily").contains(new Article("Programming Daily", false)));
+        assertTrue(groupedArticles.get("The Code").contains(new Article("The Code", false)));
+    }
+
+    @Test
+    public void givenListOfArticles_shouldSplitListInTwoCategories_usingTeeing() {
+        List<Long> countedArticles = articles.stream().collect(Collectors.teeing(
+                Collectors.filtering(article -> article.getTarget().equals("Baeldung"), Collectors.counting()),
+                Collectors.filtering(article -> !article.getTarget().equals("Baeldung"), Collectors.counting()),
+                List::of));
+
+        assertEquals(2, (long) countedArticles.get(0));
+        assertEquals(2, (long) countedArticles.get(1));
+    }
+
+    @Test
+    public void givenListOfArticles_shouldSplitListInTwoOVerlappingCategories_usingTeeing() {
+        List<List<Article>> groupedArticles = articles.stream().collect(Collectors.teeing(
+                Collectors.filtering(article -> article.getTarget().equals("Baeldung"), Collectors.toList()),
+                Collectors.filtering(Article::isFeatured, Collectors.toList()),
+                List::of));
+
+        assertEquals(2, groupedArticles.get(0).size());
+        assertEquals(1, groupedArticles.get(1).size());
+
+        assertTrue(groupedArticles.get(0).containsAll(Lists.newArrayList(
```

Comment:
This use of `assertTrue` indicates a weak assertion. Rather than use an assertion that itself checks the contents of a list and reports on the differences, we've written the logic of the assertion ourselves and just done `assertTrue`. If this assertion false, you'll get `was false, should be true`, which doesn't tell you anything about the data being asserted.

I recommend switching to AssertJ for these assertions - `assertThat(groupedArticles.get(0)).containsExactly(...)`

Reply:
Ok, I don't like mixing assertions so I migrated all of them to assertj.